db/import.go

```diff
@@ -263,13 +259,30 @@ func (db *Database) importDoc(docid string, body Body, isDelete bool, existingDo
 			bodyWithoutSpecialProps, wasStripped = stripSpecialProperties(body)
 			rawBodyForRevID, err = base.JSONMarshalCanonical(bodyWithoutSpecialProps)
 			if err != nil {
-				return nil, nil, nil, err
+				return nil, nil, false, nil, err
+			}
+		}
+
+		// If the body is the same then this import was triggered by an xattr change meaning we want to skip the below
+		// new revision handling
+		if !doc.BodyChanged(existingDoc.Body) {
+			doc.RemoveBody()
+			newDoc.UpdateBody(body)
+			if !wasStripped && !isDelete {
+				newDoc._rawBody = rawBodyForRevID
 			}
+			newDoc.RevID = doc.CurrentRev
+
+			return newDoc, nil, true, updatedExpiry, nil
 		}
```

Comment:
Wondering if this will be more maintainable to only regenerate rev id when the body changes instead. Otherwise we have 2 blocks of the same logic to keep in sync for both cases.

Reply:
Yeah that actually makes more sense imo.

Reply:
We do still need to update newDoc.RevID though.

Reply:
Fair enough, but I think can still be covered by an else

Reply:
Yeah sure. Just pointing it out :)